tensorboard/webapp/app_routing/deep_link_provider.ts

```diff
@@ -20,11 +20,30 @@ import {State} from '../app_state';
 
 import {SerializableQueryParams} from './types';
 
+/**
+ * The `DeepLinkProvider` interface standardizes the relationship between the
+ * app state and the URL parameters.  It is expected that when either an update
+ * to the URL parameters should change app state, or, in the other direction,
+ * when an update to the state should cause a change in the URL parameters,
+ * that logic should be invoked through a `DeepLinkProvider`.
```

Comment:
While statement is correct, I think it is more correct to say below:
```
TensorBoard rotuer considers URL as a view where a subset of state gets projected onto (like a HTML input element) and the DeepLinkProvider is analogous to a Container of the view. Since the state to be reflected onto the URL bar differs by a route, we have a DeepLinkProvider per route.
```